src/pygeoprocessing/geoprocessing.py

```diff
@@ -2505,6 +2517,9 @@ def convolve_2d(
             over any areas except nodata holes, in this case the resulting
             values in these areas will be nonsensical numbers, perhaps
             numerical infinity or NaNs.
+                Setting this value to ``True`` is similar to the effect
+            created by setting ``mode`` to ``"valid"`` in Numpy's
+            convolve function: https://numpy.org/doc/stable/reference/generated/numpy.convolve.html
```

Comment:
As I'm understanding it this conflicts with the description above... the docs for `"valid"` say _"The convolution product is only given for points where the signals overlap completely."_ So, the output of `numpy.convolve(..., mode="valid")` would be smaller than the input array, because edges don't overlap completely and so they're excluded.  But the output of `convolve_2d(..., ignore_nodata_and_edges=True)` is the same size as the input. I don't think there's an analogous option in numpy: `mode="same"` doesn't account for boundary effects.

Reply:
oh you're right! that's not a comparable functionality at all. okay i'm going to just delete this part.